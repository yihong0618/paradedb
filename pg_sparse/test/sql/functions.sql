-- SELECT '[1,2,3]'::svector + '[4,5,6]';
-- SELECT '[3e38]'::svector + '[3e38]';
-- SELECT '[1,2,3]'::svector - '[4,5,6]';
-- SELECT '[-3e38]'::svector - '[3e38]';
-- SELECT '[1,2,3]'::svector * '[4,5,6]';
-- SELECT '[1e37]'::svector * '[1e37]';
-- SELECT '[1e-37]'::svector * '[1e-37]';

-- SELECT svector_dims('[1,2,3]');

-- SELECT round(svector_norm('[1,1]')::numeric, 5);
-- SELECT svector_norm('[3,4]');
-- SELECT svector_norm('[0,1]');
-- SELECT svector_norm('[3e37,4e37]')::real;

-- SELECT l2_distance('[0,0]', '[3,4]');
-- SELECT l2_distance('[0,0]', '[0,1]');
-- SELECT l2_distance('[1,2]', '[3]');
-- SELECT l2_distance('[3e38]', '[-3e38]');

-- SELECT inner_product('[1,2]', '[3,4]');
-- SELECT inner_product('[1,2]', '[3]');
-- SELECT inner_product('[3e38]', '[3e38]');

-- SELECT cosine_distance('[1,2]', '[2,4]');
-- SELECT cosine_distance('[1,2]', '[0,0]');
-- SELECT cosine_distance('[1,1]', '[1,1]');
-- SELECT cosine_distance('[1,0]', '[0,2]');
-- SELECT cosine_distance('[1,1]', '[-1,-1]');
-- SELECT cosine_distance('[1,2]', '[3]');
-- SELECT cosine_distance('[1,1]', '[1.1,1.1]');
-- SELECT cosine_distance('[1,1]', '[-1.1,-1.1]');
-- SELECT cosine_distance('[3e38]', '[3e38]');

-- SELECT l1_distance('[0,0]', '[3,4]');
-- SELECT l1_distance('[0,0]', '[0,1]');
-- SELECT l1_distance('[1,2]', '[3]');
-- SELECT l1_distance('[3e38]', '[-3e38]');

-- SELECT sum(v) FROM unnest(ARRAY['[1,2,3]'::svector, '[3,5,7]']) v;
-- SELECT sum(v) FROM unnest(ARRAY['[1,2,3]'::svector, '[3,5,7]', NULL]) v;
-- SELECT sum(v) FROM unnest(ARRAY[]::svector[]) v;
-- SELECT sum(v) FROM unnest(ARRAY['[1,2]'::svector, '[3]']) v;
-- SELECT sum(v) FROM unnest(ARRAY['[3e38]'::svector, '[3e38]']) v;
