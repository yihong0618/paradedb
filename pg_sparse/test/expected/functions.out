SELECT '[1,2,3]'::svector + '[4,5,6]';
 ?column? 
----------
 [5,7,9]
(1 row)

SELECT '[3e38]'::svector + '[3e38]';
ERROR:  value out of range: overflow
SELECT '[1,2,3]'::svector - '[4,5,6]';
  ?column?  
------------
 [-3,-3,-3]
(1 row)

SELECT '[-3e38]'::svector - '[3e38]';
ERROR:  value out of range: overflow
SELECT '[1,2,3]'::svector * '[4,5,6]';
 ?column?  
-----------
 [4,10,18]
(1 row)

SELECT svector_dims('[1,2,3]');
 svector_dims 
--------------
            3
(1 row)

SELECT round(svector_norm('[1,1]')::numeric, 5);
  round  
---------
 1.41421
(1 row)

SELECT svector_norm('[3,4]');
 svector_norm 
--------------
            5
(1 row)

SELECT svector_norm('[0,1]');
 svector_norm 
--------------
            1
(1 row)

SELECT svector_norm('[3e37,4e37]')::real;
 svector_norm 
--------------
        5e+37
(1 row)

SELECT l2_distance('[0,0]', '[3,4]');
 l2_distance 
-------------
           5
(1 row)

SELECT l2_distance('[0,0]', '[0,1]');
 l2_distance 
-------------
           1
(1 row)

SELECT l2_distance('[1,2]', '[3]');
ERROR:  different svector dimensions 2 and 1
SELECT l2_distance('[3e38]', '[-3e38]');
 l2_distance 
-------------
    Infinity
(1 row)

SELECT l2_distance('[0,0,0,1]', '[3,4,0,2]');
    l2_distance     
--------------------
 5.0990195135927845
(1 row)

SELECT inner_product('[1,2]', '[3,4]');
 inner_product 
---------------
            11
(1 row)

SELECT inner_product('[1,2]', '[3]');
ERROR:  different svector dimensions 2 and 1
SELECT inner_product('[3e38]', '[3e38]');
 inner_product 
---------------
      Infinity
(1 row)

SELECT inner_product('[0,0,0,1]', '[3,4,0,2]');
 inner_product 
---------------
             2
(1 row)

SELECT cosine_distance('[1,2]', '[2,4]');
 cosine_distance 
-----------------
               0
(1 row)

SELECT cosine_distance('[1,2]', '[0,0]');
 cosine_distance 
-----------------
             NaN
(1 row)

SELECT cosine_distance('[1,1]', '[1,1]');
 cosine_distance 
-----------------
               0
(1 row)

SELECT cosine_distance('[1,0]', '[0,2]');
 cosine_distance 
-----------------
               1
(1 row)

SELECT cosine_distance('[1,1]', '[-1,-1]');
 cosine_distance 
-----------------
               2
(1 row)

SELECT cosine_distance('[1,2]', '[3]');
ERROR:  different svector dimensions 2 and 1
SELECT cosine_distance('[1,1]', '[1.1,1.1]');
 cosine_distance 
-----------------
               0
(1 row)

SELECT cosine_distance('[1,1]', '[-1.1,-1.1]');
 cosine_distance 
-----------------
               2
(1 row)

SELECT cosine_distance('[3e38]', '[3e38]');
 cosine_distance 
-----------------
             NaN
(1 row)

SELECT cosine_distance('[0,0,0,1]', '[3,4,0,2]');
  cosine_distance   
--------------------
 0.6286093236458963
(1 row)

SELECT l1_distance('[0,0]', '[3,4]');
 l1_distance 
-------------
           7
(1 row)

SELECT l1_distance('[0,0]', '[0,1]');
 l1_distance 
-------------
           1
(1 row)

SELECT l1_distance('[1,2]', '[3]');
ERROR:  different svector dimensions 2 and 1
SELECT l1_distance('[3e38]', '[-3e38]');
 l1_distance 
-------------
    Infinity
(1 row)

